drizzlepac/haputils/catalog_utils.py

```diff
@@ -813,12 +813,16 @@ def __init__(self, image, param_dict, param_dict_qc, diagnostic_mode, tp_sources
         scale = max(self.param_dict['scale'], self_scale)
         self.tp_masks = None
         if not self.image.blank:
-            self.tp_masks = make_wht_masks(self.image.wht_image, self.image.inv_footprint_mask,
-                                           scale=scale,
-                                           sensitivity=self.param_dict['sensitivity'],
-                                           kernel=(self.param_dict['region_size'],
-                                                   self.param_dict['region_size']))
-
+            self.tp_masks = make_wht_masks(
+                self.image.wht_image, 
+                self.image.inv_footprint_mask,
+                scale=scale,
+                sensitivity=self.param_dict['sensitivity'],
+                kernel=(self.param_dict['region_size'],
+                self.param_dict['region_size'])
+                )
+
+    # ? Unclear why these are here?
```

Comment:
These methods (where the method body is "pass") are present in the base class as a way to indicate these functions need to be present and fully defined in the child classes.